### ISSUE TITLE
Docker fails to start on Windows 10 Enterprise

### ISSUE THREAD

**Issue by mrdaemon** [docker/for-win#972]
### Expected behavior

Docker starts

### Actual behavior

Docker doesn't complains that hardware virtualization must be enabled.
It super is.

### Information
Diagnostic id: C8876D51-C0BD-4084-ABCC-EC253699359C/2017-08-02_20-37-41

- VT-x and VT-d are both enabled in firmware
- hyper-v is install and functions
- Virtualization: enabled shows up in Task Manager

### Steps to reproduce the behavior

  1. start docker
  2. Message pops up

**Comment by jasonbivins** [comment]
Hi @mrdaemon   It looks like the error is popping up when we try to create the MobyLinuxVM - let's clear it out and try to create it manually to see if we get a more specific error.  

Run these steps from an elevated PS prompt:
1. Stop-service com.docker.service
2. cd to C:\Program Files\Docker\Docker\resources
3. .\MobyLinux.ps1 -Destroy
4. check that the VHDX file (C:\Users\Public\Public Documents\Hyper-V\Virtual hard 
    disks\MobyLinuxVM.vhdx) is removed
5. .\MobyLinux.ps1 -Create

**Comment by mrdaemon** [comment]
I have a different default location for Hyper-V disks, but followed the instructions accordingly

```
C:\Program Files\Docker\Docker\resources> .\MobyLinux.ps1 -destroy
Script started at 15:29:57.492
Module loaded at 15:29:58.821
VM MobyLinuxVM does not exist
Destroying Switch DockerNAT...
Removing VM MobyLinuxVM...
Delete VHD D:\vhd\MobyLinuxVM.vhdx
C:\Program Files\Docker\Docker\resources> .\MobyLinux.ps1 -create
Script started at 15:30:23.059
Module loaded at 15:30:23.114
Creating Switch: DockerNAT...
Switch created.
Set IP address on switch
Creating VM MobyLinuxVM...
Setting CPUs to 2 and Memory to 2048 MB
Creating dynamic VHD: D:\vhd\MobyLinuxVM.vhdx
Attach VHD D:\vhd\MobyLinuxVM.vhdx
Connect Internal Switch DockerNAT
Attach DVD .\mobylinux.iso
Disabled Interface de services d’invité
Enabled Pulsation
Disabled Échange de paires clé-valeur
Enabled Arrêt
Enabled Synchronisation date/heure
Disabled VSS
VM created.
```
It still fails with the same error message however.

I have taken the liberty of uploading a new diagnostic after doing this, the id is: C8876D51-C0BD-4084-ABCC-EC253699359C/2017-08-03_15-32-27

**Comment by mrdaemon** [comment]
As an update, I tried starting the MobyLinux VM and it failed with a Resource Allocation error, which I started getting on _some_ other Virtual Machines in my Hyper-V setup as well.

As a last ditch effort to eliminate potential causes, I removed and reinstalled the Hyper-V feature entirely, removed and recreated the MobyLinux VM using the powershell helper script and it started working correctly.

I'm not sure what the cause was at this point but this clearly points to something Odd or Subtly Broken in my Hyper-V install. In any case this seems to have resolved the issue. 

If there is any further information I can provide, do ask
